sys/arduino/base.cpp

```diff
@@ -58,3 +61,40 @@ void delay(unsigned long msec)
 {
     xtimer_usleep(1000 * msec);
 }
+
+#ifdef ADC_NUMOF
+int analogRead(int adc_pin)
+{
+    int adc_value;
+
+    /* State of ADC channel.
+     * 0: Not initialized
+     * 1: Successfully initialized
+     */
+    static int adc_state[ADC_NUMOF];
```

Comment:
for style reasons, I would prefer to define this variable outside of the function, its easier to read that way...

Also: IMHO it does not make sense to waste a lot of memory here by allocating a full `int` for each ADC line. How about a bitfield instead? Using a `uint16_t` should be ok, `uint32_t` would be pretty safe, or even better the type could be defined depending on the value `ADC_NUMOF`...